politeiad/backend/gitbe/cms.go

```diff
@@ -0,0 +1,1183 @@
+		return err
+	}
+
+	// git pull --ff-only --rebase
+	err = g.gitPull(g.unvetted, true)
+	if err != nil {
+		return err
+	}
+
+	// git checkout -b timestamp_flushvotes
+	branch := strconv.FormatInt(time.Now().Unix(), 10) + "_flushvotes"
+	_ = g.gitBranchDelete(g.unvetted, branch) // Just in case
+	err = g.gitNewBranch(g.unvetted, branch)
+	if err != nil {
+		return err
+	}
+
+	// closure to handle unwind if needed
+	var errUnwind error
+	defer func() {
+		if errUnwind == nil {
+			return
+		}
+		err := g.flushJournalsUnwind(branch)
+		if err != nil {
+			log.Errorf("flushJournalsUnwind: %v", err)
+		}
+	}()
+
+	// Flush journals
+	files, err := g._flushDCCVotesJournals()
+	if err != nil {
+		errUnwind = err
+		return err
+	}
+
+	if len(files) == 0 {
+		log.Info("flushVotesJournals: nothing to do")
+		err = g.flushJournalsUnwind(branch)
+		if err != nil {
+			log.Errorf("flushJournalsUnwind: %v", err)
+		}
+		return nil
+	}
+
+	// git add journals
+	commitMessage := "Flush vote journals.\n\n"
+	for _, v := range files {
+		err = g.gitAdd(g.unvetted, v)
+		if err != nil {
+			errUnwind = err
+			return err
+		}
+
+		s := strings.Split(v, string(os.PathSeparator))
+		if len(s) == 0 {
+			commitMessage += "ERROR: " + v + "\n"
+		} else {
+			commitMessage += s[0] + "\n"
+		}
+	}
+
+	// git commit
+	err = g.gitCommit(g.unvetted, commitMessage)
+	if err != nil {
+		errUnwind = err
+		return err
+	}
+
+	// git rebase master
+	err = g.rebasePR(branch)
+	if err != nil {
+		errUnwind = err
+		return err
+	}
+
+	return nil
+}
+func (g *gitBackEnd) cmsPluginJournalFlusher() {
+	// XXX make this a single PR instead of 2 to save some git time
+	err := g.flushDCCVoteJournals()
+	if err != nil {
+		log.Errorf("cmsPluginVoteFlusher: %v", err)
+	}
+}
+
+func (g *gitBackEnd) pluginStartDCCVote(payload string) (string, error) {
+	vote, err := cmsplugin.DecodeStartVote([]byte(payload))
+	if err != nil {
+		return "", fmt.Errorf("DecodeStartVote %v", err)
+	}
+
+	// Verify vote bits are somewhat sane
+	for _, v := range vote.Vote.Options {
+		err = _validateCMSVoteBit(vote.Vote.Options, vote.Vote.Mask, v.Bits)
+		if err != nil {
+			return "", fmt.Errorf("invalid vote bits: %v", err)
+		}
+	}
+
+	// Verify dcc exists
+	tokenB, err := util.ConvertStringToken(vote.Vote.Token)
+	if err != nil {
+		return "", fmt.Errorf("ConvertStringToken %v", err)
+	}
+	token := vote.Vote.Token
+
+	if !g.vettedPropExists(token) {
+		return "", fmt.Errorf("unknown proposal: %v", token)
+	}
+
+	// Make sure vote duration is within min/max range
+	// XXX calculate this value for testnet instead of using hard coded values.
+	if vote.Vote.Duration < cmsplugin.VoteDurationMin ||
+		vote.Vote.Duration > cmsplugin.VoteDurationMax {
+		// XXX return a user error instead of an internal error
+		return "", fmt.Errorf("invalid duration: %v (%v - %v)",
+			vote.Vote.Duration, cmsplugin.VoteDurationMin,
+			cmsplugin.VoteDurationMax)
+	}
+
+	// 1. Get best block
+	bb, err := bestBlock()
```

Comment:
The best block function uses the decred plugin settings.  I'm guessing this is still going to work because of the fact that cms uses the decred plugin, but you have the dcrdata setting redefined for cms plugin, which is unnecessary.